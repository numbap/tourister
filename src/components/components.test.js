import React from 'react';
import ReactDOM from 'react-dom';
import uuid from 'uuid'
import getAppStore from '../redux/store/store';
import {Provider} from 'react-redux'
import ReactShallowRenderer from 'react-test-renderer/shallow'
import LoginButton from './login/login'
import LocationRowForm from './locations/locationrowform'
import Locations from './locations/locations'
import HomePage from './homepage/homepage'
import HomeRowForm from './homepage/homerowform'
import Header from './header/header'
import Footer from './footer/footer'

const store = getAppStore();

test('LoginButton render test', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
      <Provider store={store}>
        <LoginButton />
      </Provider>);
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })


test('LocationRowForm render test', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
      <Provider store={store}>
        <LocationRowForm />
      </Provider>);
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

test('Locations render test', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
      <Provider store={store}>
        <Locations />
      </Provider>);
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

test('HomeRowForm render test', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
      <Provider store={store}>
        <HomeRowForm />
      </Provider>);
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

test('HomePage render test', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
      <Provider store={store}>
        <HomePage />
      </Provider>);
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

test('Header render test', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
      <Provider store={store}>
        <Header />
      </Provider>);
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

test('Footer render test', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
      <Provider store={store}>
        <Footer />
      </Provider>);
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })