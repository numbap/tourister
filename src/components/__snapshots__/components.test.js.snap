// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render test 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Footer />
</Context.Provider>
`;

exports[`Header render test 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Header />
</Context.Provider>
`;

exports[`HomePage render test 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Memo(ConnectFunction) />
</Context.Provider>
`;

exports[`HomeRowForm render test 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <HomeRowForm />
</Context.Provider>
`;

exports[`LocationRowForm render test 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <LocationRowForm />
</Context.Provider>
`;

exports[`Locations render test 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Memo(ConnectFunction) />
</Context.Provider>
`;

exports[`LoginButton render test 1`] = `
<Context.Provider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <LoginButton />
</Context.Provider>
`;
